require.Equal panics when using string type alias

I'm trying to compare IDs of package github.com/graph-gophers/graphql-go. These IDs are actually just a type alias for strings:
```go
// ID represents GraphQL's "ID" scalar type. A custom type may be used instead.
type ID string
```

However if these IDs are not equal, require.Equal panics:
```go
func TestResolver(t *testing.T) {
	require.Equal(t, graphql.ID("2015"), graphql.ID("1915"))
}
```
```bash
=== RUN   TestResolver
--- FAIL: TestResolver (0.00s)
panic: interface conversion: interface {} is graphql.ID, not string [recovered]
	panic: interface conversion: interface {} is graphql.ID, not string

goroutine 148 [running]:
testing.tRunner.func1(0xc4203ac1e0)
	/usr/local/go/src/testing/testing.go:742 +0x567
panic(0x1a60400, 0xc420280440)
	/usr/local/go/src/runtime/panic.go:502 +0x24a
github.com/SpaceHeroGuide/wertungsheft/vendor/github.com/stretchr/testify/assert.diff(0x1a56c80, 0x1c484e0, 0x1a56c80, 0x1c484f0, 0x0, 0x0)
	/Users/matthias/Golang/src/github.com/SpaceHeroGuide/wertungsheft/vendor/github.com/stretchr/testify/assert/assertions.go:1352 +0x4ba
github.com/SpaceHeroGuide/wertungsheft/vendor/github.com/stretchr/testify/assert.Equal(0x44a0188, 0xc4203ac1e0, 0x1a56c80, 0x1c484e0, 0x1a56c80, 0x1c484f0, 0x0, 0x0, 0x0, 0xc4203ac1e0)
	/Users/matthias/Golang/src/github.com/SpaceHeroGuide/wertungsheft/vendor/github.com/stretchr/testify/assert/assertions.go:338 +0x2f8
github.com/SpaceHeroGuide/wertungsheft/vendor/github.com/stretchr/testify/require.Equal(0x1c508e0, 0xc4203ac1e0, 0x1a56c80, 0x1c484e0, 0x1a56c80, 0x1c484f0, 0x0, 0x0, 0x0)
	/Users/matthias/Golang/src/github.com/SpaceHeroGuide/wertungsheft/vendor/github.com/stretchr/testify/require/require.go:157 +0xc0
github.com/SpaceHeroGuide/wertungsheft/vendor/github.com/stretchr/testify/require.(*Assertions).Equal(0xc42036ae68, 0x1a56c80, 0x1c484e0, 0x1a56c80, 0x1c484f0, 0x0, 0x0, 0x0)
	/Users/matthias/Golang/src/github.com/SpaceHeroGuide/wertungsheft/vendor/github.com/stretchr/testify/require/require_forward.go:130 +0xe3
github.com/SpaceHeroGuide/wertungsheft/pkg/api/tournament_test.TestResolver(0xc4203ac1e0)
	/Users/matthias/Golang/src/github.com/SpaceHeroGuide/wertungsheft/pkg/api/tournament/tournament_test.go:90 +0x163
testing.tRunner(0xc4203ac1e0, 0x1b6a7a0)
	/usr/local/go/src/testing/testing.go:777 +0x16e
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:824 +0x565
FAIL	github.com/SpaceHeroGuide/wertungsheft/pkg/api/tournament	2.198s
```

If you change the IDs so they are equal, the panic does not occur.

--- a/assert/assertions.go
+++ b/assert/assertions.go
@@ -1345,7 +1345,7 @@ func diff(expected interface{}, actual interface{}) string {
 	}
 
 	var e, a string
-	if ek != reflect.String {
+	if et != reflect.TypeOf("") {
 		e = spewConfig.Sdump(expected)
 		a = spewConfig.Sdump(actual)
 	} else {

--- a/assert/assertions_test.go
+++ b/assert/assertions_test.go
@@ -175,6 +175,8 @@ func TestIsType(t *testing.T) {
 
 }
 
+type myType string
+
 func TestEqual(t *testing.T) {
 
 	mockT := new(testing.T)
@@ -200,11 +202,17 @@ func TestEqual(t *testing.T) {
 	if !Equal(mockT, uint64(123), uint64(123)) {
 		t.Error("Equal should return true")
 	}
+	if !Equal(mockT, myType("1"), myType("1")) {
+		t.Error("Equal should return true")
+	}
 	if !Equal(mockT, &struct{}{}, &struct{}{}) {
 		t.Error("Equal should return true (pointer equality is based on equality of underlying value)")
 	}
 	var m map[string]interface{}
 	if Equal(mockT, m["bar"], "something") {
+		t.Error("Equal should return false")
+	}
+	if Equal(mockT, myType("1"), myType("2")) {
 		t.Error("Equal should return false")
 	}
 }

